test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/PreserveLogsRule.java

```diff
@@ -0,0 +1,17 @@
+package com.datastax.oss.driver.api.testinfra.ccm;
+
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+public class PreserveLogsRule extends TestWatcher {
```

Comment:
Question - why not to make it configurable from environment variable ? So that it could be easily set for whole run.

Reply:
IIRC this was the simplest way to have the rule react only in case of failure. The goal was to skip logs of passing runs.
Global switch would be nice for running locally, but I'm not sure if we want CI to keep regularly uploading unnecessary log files.

Reply:
We definitely would want to store ccm logs on test failures ok cicd.

Reply:
Let's make it global flag that controls whether we keep ccm logs after the test or not.